crates/database/src/orders.rs

```diff
@@ -331,7 +399,10 @@ o.is_liquidity_order,
     WHERE o.uid = p.order_uid
     ORDER BY p.block_number DESC, p.log_index DESC
     LIMIT 1
-), true)) AS presignature_pending
+), true)) AS presignature_pending,
+array(Select target from interactions p where p.order_uid = o.uid order by p.index) as pre_interactions_tos,
+array(Select value from interactions p where p.order_uid = o.uid order by p.index) as pre_interactions_values,
+array(Select data from interactions p where p.order_uid = o.uid order by p.index) as pre_interactions_data
```

Comment:
Regarding ☝️, maybe:
```sql
array(SELECT (i.target, i.value, i.data) FROM interactions i WHERE i.order_uid = o.uid ORDER BY p.index) AS pre_interactions
```

Not sure if this is possible though...

Reply:
I tried this one before but always tried to retrieve an Interaction struct. 
But if retrieving a triple (Address, U256, Vec<u8>) Vec, then it works.
I mean it's not the beautiful solution I was looking for, but much much better already. 

(I tried always to get the struct from json objects, but this is something that is not really supported from sqlx)

Thank you! ❤️